### PR TITLE
Support pass kwargs to cogvideox custom attention processor

### PR DESCRIPTION
**Pull request by huanngzh**
# What does this PR do?

Pass kwargs to custom attention processor is preferable when implementing layout control, attention replacing or introducing side effect to attention computation. Now this feature is supported by sd, sdxl, FLUX, sd3 series, but not CogVideoX. This PR fix the problem, and may support future training-free experiments in CogVideoX series models.

Example code:
```Python
class CustomCogVideoXAttnProcessor2_0:
    def __init__(self, **custom_args):
        super().__init__()
        # some initialization
    
    def __call__(self, ..., added_argu=None) -> torch.FloatTensor:
        # custom attention forward pass

custom_processor = CustomCogVideoXAttnProcessor2_0(**custom_args)
pipe.transformer.set_attn_processor(custom_processor) # cogvideox pipeline
attention_kwargs = {"added_argu": 100} # prepare passing added_argu to attention processor
pipe.transformer(..., attention_kwargs=attention_kwargs) # custom transformer forward pass
```

## Before submitting
- [ ] This PR fixes a typo or improves the docs (you can dismiss the other checks if that's the case).
- [x] Did you read the [contributor guideline](https://github.com/huggingface/diffusers/blob/main/CONTRIBUTING.md)?
- [x] Did you read our [philosophy doc](https://github.com/huggingface/diffusers/blob/main/PHILOSOPHY.md) (important for complex PRs)?
- [ ] Was this discussed/approved via a GitHub issue or the [forum](https://discuss.huggingface.co/c/discussion-related-to-httpsgithubcomhuggingfacediffusers/63)? Please add a link to it if that's the case.
- [ ] Did you make sure to update the documentation with your changes? Here are the
      [documentation guidelines](https://github.com/huggingface/diffusers/tree/main/docs), and
      [here are tips on formatting docstrings](https://github.com/huggingface/diffusers/tree/main/docs#writing-source-documentation).
- [ ] Did you write any new necessary tests?

## Who can review?

Anyone in the community is free to review the PR once the tests have passed. Feel free to tag
members/contributors who may be interested in your PR.

Hello @yiyixuxu @sayakpaul, I just want to do the same to CogVideoX as to FLUX in https://github.com/huggingface/diffusers/pull/9517 and to SD3 in https://github.com/huggingface/diffusers/pull/9818

### CODE REVIEW

File: src/diffusers/models/attention_processor.py [hunk 1]
```diff
@@ -2813,6 +2813,8 @@ def __call__(
         encoder_hidden_states: torch.Tensor,
         attention_mask: Optional[torch.Tensor] = None,
         image_rotary_emb: Optional[torch.Tensor] = None,
+        *args,
```
Comment (yiyixuxu) [inline src/diffusers/models/attention_processor.py:2816]: we don't need to accept more arguments here, no? you are going to use the custom attention processor anyway

Comment (a-r-r-o-w) [reply src/diffusers/models/attention_processor.py:2816]: Maybe should add attention_kwargs by default whenever adding a new model for use cases like this. Regional prompting, conditional attention masking, etc. seem to be significantly used by power-users of UIs, so if that's okay with you, will do newer model additions with this in mind

Comment (huanngzh) [reply src/diffusers/models/attention_processor.py:2816]: Perhaps accepting more arguments is not necessary here, because it will be detected and intercepted by the Attention forward. If agreed, I will delete these two lines.

Comment (a-r-r-o-w) [reply src/diffusers/models/attention_processor.py:2816]: I think what we mean here is to keep `**kwargs` but remove `*args`. Your changes look good to merge except for the two occurences of `*args` in the attention processors

Comment (huanngzh) [reply src/diffusers/models/attention_processor.py]: Have removed it.

**Review by a-r-r-o-w — APPROVED** [review]
Thanks!

File: src/diffusers/models/attention_processor.py [hunk 2]
```diff
@@ -2813,6 +2813,7 @@ def __call__(
         encoder_hidden_states: torch.Tensor,
         attention_mask: Optional[torch.Tensor] = None,
         image_rotary_emb: Optional[torch.Tensor] = None,
+        **kwargs,
```
Comment (yiyixuxu) [inline src/diffusers/models/attention_processor.py:2816]: oh I just saw the example code here you provided 

https://github.com/huggingface/diffusers/pull/10456#issue-2769170591

your custom attention processor should not inherit from the official one, so I don't think the **kwargs here is needed, no

Comment (huanngzh) [reply src/diffusers/models/attention_processor.py:2816]: Hi, I have updated my example code of custom attention processor.

**Review by yiyixuxu — APPROVED** [review]
thanks!
